### ISSUE TITLE
jooby can't parse nested params without a default no-arg constructor

### ISSUE THREAD

**Issue by sahilm** [jooby-project/jooby#1093]
With a jooby app like this:

```java
public class App extends Jooby {

    {
        get("/search", (req, rsp) -> {
            final Profile profile = req.params(Profile.class);
            System.out.println(profile);
        });
    }

    public static void main(final String[] args) {
        run(App::new, args);
    }
}
```

```java
public class Profile {
    public final String name;
    public final int age;
    public final Address address;

    @Inject
    public Profile(final String name, final int age, final Address address) {
        this.name = name;
        this.age = age;
        this.address = address;
    }
}
```

```java
public class Address {
    public final String country;
    public final String city;

    @Inject
    public Address(final String country, final String city) {
        this.country = country;
        this.city = city;
    }
}
```
The following GET request:

```bash
curl -g 'http://localhost:8080/search?name=Foo&age=12&address[city]=BA&address[country]=ABCD'
```
Fails with:
```
message: Bad Request(400): Failed to parse parameter 'address' to 'com.sahilm.Address'
```

Making all fields non-final and removing the constructor fixes things.

Code to reproduce this issue can be found in [sahilm/jooby-failed-to-parse-nested-param](https://github.com/sahilm/jooby-failed-to-parse-nested-param). I'm on jooby `v1.3.0`. Let me know if you need any further information or help. Thanks!

**Comment by jknack** [comment]
Not sure yet how to fix it, but the workaround if you need to keep your immutable object is to annotated the constructor arguments with `Named`

```
public class Address {
    public final String country;
    public final String city;

    @Inject
    public Address(final @Named("address[country]") String country, final @Named("address[city]") String city) {
        this.country = country;
        this.city = city;
    }
}
```

**Comment by sahilm** [comment]
That works! Thanks.

**Comment by jknack** [comment]
Fixed in 2.0.0
